demos/social_distance_demo/cpp/src/geodist.cpp

```diff
@@ -0,0 +1,312 @@
+}
+
+bool lineContainsPoint(const Line2d& l, const cv::Point2d& p) {
+    const cv::Point2d& a = l.first;
+    const cv::Point2d& b = l.second;
+
+    if (std::isnan(a.x) || std::isnan(a.y) || std::isnan(b.x) || std::isnan(b.y) ||
+        std::isnan(p.x) || std::isnan(p.x)) {
+        return false;
+    }
+
+    double crossproduct = (b.x - a.x) * (p.y - a.y) - (p.x - a.x) * (b.y - a.y);
+    double dotproduct = (p.x - a.x) * (b.x - a.x) + (p.y - a.y) * (b.y - a.y);
+
+    if (std::fabs(crossproduct/dotproduct) > 0.0000001) {
+        return false;
+    }
+
+    if (dotproduct < 0) {
+        return false;
+    }
+
+    double sqlen = (b.x - a.x) * (b.x - a.x) + (b.y - a.y) * (b.y - a.y);
+    if (dotproduct > sqlen) {
+        return false;
+    }
+
+    return true;
+}
+
+// euclidean, alert, distance
+std::tuple<bool, bool, double> euclideanDistance(const Line2d &AB, const Line2d &CD,
+                                                  unsigned minIter, double coef) {
+    cv::Point2d Z1 = lineCentroid(AB);
+    cv::Point2d Z2 = lineCentroid(CD);
+
+    double minDist = (lineLength(AB) * minIter) * 0.8;
+    double distance = getDistance(Z1, Z2) * coef;
+    if (distance < lineLength(AB)) {
+        return std::make_tuple(true, false, distance);
+    }
+
+    bool alert = distance < minDist ? true : false;
+
+    return std::make_tuple(true, alert, distance);
+}
+
+double getDistance(const cv::Point2d &PF, const cv::Point2d &E, const cv::Point2d &F,
+                    cv::Point2d &za, cv::Point2d &zb, const cv::Point2d &zLimitA, const cv::Point2d &zLimitB,
+                    double inf1Dir, unsigned initIter = 1, double coef = 1) {
+    double _inf = inf1Dir;
+    double inf = inf1Dir * -1;
+    bool zaOverZLimit = false;
+    Line2d PFE = getLine(PF, E);
+    Line2d PFF = getLine(PF, F);
+    double lenEF = getDistance(E, F);
+    unsigned cnt = initIter;
+
+    while (!zaOverZLimit) {
+        if (inf > 0) {
+            // inf = 9999
+            // _inf = -9999
+            double lenZbF = getDistance(zb, F);
+            cv::Point2d fProj(F.x, F.y + lenZbF);
+            cv::Point2d eProj(E.x, F.y + lenZbF + lenEF);
+
+            double projA, projK;
+            std::tie(projA, projK) = getLineComponent(eProj, fProj);
+            cv::Point2d AUX(getX(F.y, projA, projK), F.y);
+
+            double auxA, auxK;
+            std::tie(auxA, auxK) = getLineComponent(zb, AUX);
+            Line2d AUXinf = getLine(cv::Point2d(_inf, getY(_inf, auxA, auxK)), AUX);
+            cv::Point2d zAUXa = lineIntersection(PFE, AUXinf);
+
+            if (zAUXa.y < zLimitA.y) {
+                zaOverZLimit = true;
+            } else {
+                cnt += 1;
+                za = zAUXa;
+                Line2d zbAUX = getLine(za, cv::Point2d(inf, za.y));
+                zb = lineIntersection(PFF, zbAUX);
+            }
+        } else {
+            // inf = - 9999
+            // _inf = 9999
```

Comment:
I think from here in from if branch these comments should be removed.

Reply:
Done.